v2/client/invoker.go

```diff
@@ -65,24 +70,20 @@ func (r *receiveInvoker) Invoke(ctx context.Context, m binding.Message, respFn p
 			}
 		}
 
-		// protocol can manual ack by the result
+		// protocol can ack itself based on the result
 		if respFn == nil {
 			if !protocol.IsACK(result) {
-				err = m.Finish(result)
 				isFinished = true
 			}
-			return
+			return m.Finish(result)
```

Comment:
this will cause finish to be called twice if !Ack

Reply:
Where would be the second time?

Reply:
Got it, in the defer